### DISCUSSION TITLE
Difficulty reading in 3D fieldset from NEMO

### DISCUSSION THREAD

**Discussion by mollyjames2** [Parcels-code/Parcels#1453] [Q&A] [answered]
Hi,
I am using the NEMO 3D tutorial to try and read in a NEMO dataset to run a 3D particle tracking simulation and I'm running into issues. to read the data i have the following code: 
`dataset_folder = "/data/proteus1/scratch/moja/projects/SANH/Leak1H"
ufiles = sorted(glob(f"{dataset_folder}/SRIL34_1HL_*1h_grid_U.nc"))
vfiles = sorted(glob(f"{dataset_folder}/SRIL34_1HL_*1h_grid_V.nc "))
wfiles = sorted(glob(f"{dataset_folder}/SRIL34_1HL_*1h_grid_W.nc"))
mesh_mask = f"{dataset_folder}/coordinates.nc"

filenames = {
    "U": {"lon": mesh_mask, "lat": mesh_mask, "depth": wfiles[0], "data": ufiles},
    "V": {"lon": mesh_mask, "lat": mesh_mask, "depth": wfiles[0], "data": vfiles},
    "W": {"lon": mesh_mask, "lat": mesh_mask, "depth": wfiles[0], "data": wfiles},
}

variables = {"U": "uoce", "V": "voce", "W": "woce"}

c_grid_dimensions = {
    "lon": "glamf",
    "lat": "gphif",
    "depth": "depthw",
    "time": "time_counter",
}
dimensions = {
    "U": c_grid_dimensions,
    "V": c_grid_dimensions,
    "W": c_grid_dimensions,
}

fieldset = FieldSet.from_nemo(filenames, variables, dimensions)`

but I am getting the error: 
`OSError: FieldSet files not found for variable V: {'lon': ['/data/proteus1/scratch/moja/projects/SANH/Leak1H/coordinates.nc'], 'lat': ['/data/proteus1/scratch/moja/projects/SANH/Leak1H/coordinates.nc'], 'depth': ['/data/proteus1/scratch/moja/projects/SANH/Leak1H/SRIL34_1HL_20210501_20210531_1h_grid_W.nc'], 'data': []}`

in my coordinates.nc the data is formatted as follows:
`netcdf coordinates {
dimensions:
        x = 324 ;
        y = 424 ;
        z = 1 ;
        time = 1 ;
variables:
        float nav_lon(y, x) ;
                nav_lon:axis = "Longitude" ;
                nav_lon:short_name = "nav_lon" ;
                nav_lon:units = "degrees_east" ;
                nav_lon:long_name = "Longitude" ;
        float nav_lat(y, x) ;
                nav_lat:axis = "Latitude" ;
                nav_lat:short_name = "nav_lat" ;
                nav_lat:units = "degrees_east" ;
                nav_lat:long_name = "Latitude" ;
        float nav_lev(z) ;
                nav_lev:axis = "Model Level" ;
                nav_lev:short_name = "nav_lev" ;
                nav_lev:units = "unitless" ;
                nav_lev:long_name = "Model Level" ;
        float e1t(y, x) ;
        float e2t(y, x) ;
        float glamt(y, x) ;
        float gphit(y, x) ;
        float e1u(y, x) ;
        float e2u(y, x) ;
        float glamu(y, x) ;
        float gphiu(y, x) ;
        float e1v(y, x) ;
        float e2v(y, x) ;
        float glamv(y, x) ;
        float gphiv(y, x) ;
        float e1f(y, x) ;
        float e2f(y, x) ;
        float glamf(y, x) ;
        float gphif(y, x) ;
`

any help or advice on how to get this data read in would be greatly beneficial - thanks in advance

**Comment by erikvansebille** [comment, marked as answer]
Thanks for asking, @mollyjames2; but I think the error message says it all, in the last few words: `[...] 'data': []}`

Your list of data-files is empty for variable `V`. Are you sure that `vfiles = sorted(glob(f"{dataset_folder}/SRIL34_1HL_*1h_grid_V.nc "))` does not give you an empty list? Do these files exist?

**Reply by mollyjames2** [reply]
thank you for your quik reply Erik - on closer inspection of the script it seems like the accidental space between .nc and " was the culprit! many thanks
